digdag-core/src/main/java/io/digdag/core/database/ThreadLocalTransactionManager.java

```diff
@@ -82,21 +140,37 @@ public void abort()
             if (handle == null) {
                 return;
             }
+            if (state == State.COMMITTED) {
+                throw new IllegalStateException("Aborting committed transaction is not allowed");
+            }
             handle.rollback();
+            state = State.ABORTED;
             // TODO Should call close()?
         }
     }
 
     @Inject
     public ThreadLocalTransactionManager(DataSource ds)
     {
+        this(ds, false);
+    }
+
+    ThreadLocalTransactionManager(DataSource ds, boolean autoAutoCommit)
+    {
         this.ds = checkNotNull(ds);
+        if (autoAutoCommit) {
+            LazyTransaction transaction = new LazyTransaction(ds, true);
```

Comment:
Creating auto-commit style transaction and storing it in Thread local should be in `begin()` method lazily?